Remotion/Data/DomainObjects/Persistence/IStorageProvider.cs

```diff
@@ -0,0 +1,31 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System.Collections.Generic;
+using Remotion.Data.DomainObjects.DataManagement;
+using Remotion.Data.DomainObjects.Mapping;
+
+namespace Remotion.Data.DomainObjects.Persistence
+{
+  public interface IStorageProvider : IReadOnlyStorageProvider
```

Comment:
yes, please XML comment. Can probably be taken from StorageProvider.
 And something that will be done once we are done with the interface switcheroo: we can drop StorageProvider (the class) and only use interfaces since there's no relevant implementation in storageprovider left that would make an abstract baseclass relevant. I think this should be a separate task though. @Frazerus please add a note to reevaluate this after we replaced the type usages with the interface in ticket RM-9127. There we can decide if inlining & droping the StorageProvider class into RdbmsProvider can be done immidiately or should be deferred into a separate task.